cogent/Makefile

```diff
@@ -50,33 +41,49 @@ build: setup-config
 	$(E) "Building..."
 	$(CABAL) build
 
-configure: install-deps
+configure:
 	$(E) "Configuring"
 	$(CABAL) configure
-	$(CABAL) build
 
 sandbox:
-	$(E) "Building in a sandbox"
+ifneq ($(wildcard .cabal-sandbox/),)
+	$(E) "Using Existing sandbox..."
+else
+	$(E) "Building in a sandbox..."
 	$(CABAL) sandbox init
 	$(CABAL) sandbox add-source ../isa-parser
+endif
 
-install-deps: sandbox
+.install-deps:
 	$(E) "Installing dependencies"
-	$(CABAL) install --only-dependencies --force-reinstalls
+	$(CABAL) install --only-dependencies
 
 doc:
-	$(E) "Docs"
+	$(E) "Building Cogent docs..."
+	$(Q) make -C doc
 
 clean:
 	$(E) "Cleaning"
-	$(Q) cabal sandbox delete -v0
+	$(Q) $(CABAL) sandbox delete -v0
 	$(Q) rm -rf .cabal-sandbox/ dist/ ../isa-parser/dist
 	$(Q) rm -rf out/
 	$(Q) rm -rf tests/include
+	$(Q) make -C doc clean
 	$(Q) find . -name "*_flymake.hs" -delete
 
-dev: configure
-	$(CABAL) install $(CABALFLAGS)
+install-global: cabal-update isa-parser .install-deps configure build .install
+	$(E) "Cogent Compiler successfully built for $(OS)."
+	$(E)
+	$(E) "Add 'cogent' compiler to your path by running:"
+	$(E) '  export PATH=$$PATH:~/.cabal/bin'
+	$(E)
+	$(E) "To enable bash auto-completion for 'cogent', please add the following to your .bashrc file:"
+	$(E) "source $(PWD)/misc/cogent_autocomplete.sh"
+	$(E)
+	$(E)
+
+install-dev: sandbox .install-deps configure build .install
```

Comment:
`install-deps` installs the dependencies only and `install` installs either `Cogent` or `isa-parser`.